src/engine/MILPEncoder.cpp

```diff
@@ -126,3 +129,85 @@ void MILPEncoder::encodeReLUConstraint( GurobiWrapper &gurobi, ReluConstraint
     terms.append( GurobiWrapper::Term( -sourceUb, Stringf( "a%u", index ) ) );
     gurobi.addLeqConstraint( terms, 0 );
 }
+
+void MILPEncoder::encodeMaxConstraint( GurobiWrapper &gurobi, MaxConstraint *max )
+{
+
+    if ( !max->isActive() )
+        return;
+
+    // y = max(x_1, x_2, ... , x_m)
+    unsigned y = max->getF();
+
+    // xs = [x_1, x_2, ... , x_m]
+    List<unsigned> xs = max->getParticipatingVariables();
+    unsigned m = xs.size();
+
+    // upper bounds of each x_i
+    double* ubs = new double[m];
+
+    // terms for Gurobi
+    List<GurobiWrapper::Term> terms;
+
+    unsigned i = 0;
+    for ( const auto &x : xs ) {
+        // add binary variable
+        gurobi.addVariable( Stringf( "a%u", x ),
```

Comment:
Naming the auxiliary variable like this would result in duplicated variable names. For instance, if we have two Max constraints Max([x1,x2], x4) and Max([x1,x3], x5), we would introduce the variable a1 twice, while we actually need two fresh variables. 
One solution is to keep track of the latest index of the auxiliary variable. Please check the encodeReLUConstraint() method above.

Reply:
will do.

Reply:
done.